backend/app/models/sample_type.rb

```diff
@@ -0,0 +1,284 @@
+# sample_types table
+class SampleType < ActiveRecord::Base
+
+  validates :name,        presence: true, uniqueness: { case_sensitive: false }
+  validates :description, presence: true
+
+  # Return all sample types.
+  #
+  # @return all sample types
+  def self.find_all
+    sql = "
+      select * from sample_types order by name
+    "
+    SampleType.find_by_sql sql
+  end
+
+  def self.find_id(id)
+    sql = "
+      select * from sample_types where id = #{id} limit 1
+    "
+    (SampleType.find_by_sql sql)[0]
+  end
+
+  # Return details for a specific sample types.
+  #
+  # @param id [Int] the id of the sample type
+  #
+  # @return the sample types
+  def self.details(id)
+    # GET FEILD TYPES
+    sql = "
+      select *, null as 'allowable_field_types'
+      from field_types ft
+      where ft.parent_class = 'SampleType' and ft.parent_id = #{id}
+      order by ft.name
+    "
+    field_types = FieldType.find_by_sql sql
+
+    # GET SAMPLE_OPTIONS FOR FIELD_TYPE == "SAMPLE"
+    # MAKES <N> CALLS TO THE DB BUT <N> SHOULD NOT BE LARGE
+    field_types.each do |ft|
```

Comment:
Let's spell out variable names instead of using the functional style.
So, `field_type` instead of `ft`